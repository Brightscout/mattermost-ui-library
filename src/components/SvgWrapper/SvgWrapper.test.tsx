import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import SvgWrapper from '.';

// Mock props data
const svgWrapperChildren = <span>{'Mock Svg'}</span>;
const svgWrapperClassName = 'svg-wrapper-class';

describe('SvgWrapper', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<SvgWrapper className={svgWrapperClassName}>{svgWrapperChildren}</SvgWrapper>);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(svgWrapperClassName)).toBeTruthy();
    });

    it('Should render the children correctly', () => {
        expect(component.contains(svgWrapperChildren)).toBeTruthy();
    });
});
