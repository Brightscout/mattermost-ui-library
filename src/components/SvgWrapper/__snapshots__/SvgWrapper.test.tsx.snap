// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgWrapper Should render correctly 1`] = `ShallowWrapper {}`;
